scvelo/tools/_em_model.py

```diff
@@ -274,7 +264,7 @@ def fit(
             "use_raw": use_raw,
         }
 
-        _write_pars(adata, _pars, add_key=self._fit_key)
+        _write_pars(adata, list(asdict(sd).values()), add_key=self._fit_key)
```

Comment:
Same as for `_read_pars`. Also, let's use the `export_results_adata` function for writing parameters and not have it part of the `fit` method.

Reply:
I used `export_results_adata`, but I noticed that `fit` does not work without writing the parameters during fitting (or at least it would be a larger refactoring, which I'd want to do in a separate commit). The parameters written to the adata object are used by the `align_dynamics` method, which is called during `fit` after writing.

`align_dynamics` also modifies the written values. Hence, `export_results_adata` currently gives different results than there are returned by `fit`.

For refactoring, I'd suggest to modify `align_dynamics` as well such that is uses `EMParams` and copy it to `_em_model.py` (currently the method is in `_em_model_core.py`). Is this approach okay or would you suggest something different?